plugins/module_utils/CertACLHelper.cs

```diff
@@ -0,0 +1,282 @@
+        [return: MarshalAs(UnmanagedType.Bool)]
+        private static extern bool CryptGetProvParam(CryptHandle safeProvHandle, CryptProvParam dwParam, SafeSecurityDescriptorPtr pbData, ref uint dwDataLen, SecurityInformationFlags dwFlags);
+
+        [DllImport("advapi32.dll", CharSet = CharSet.Unicode, SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        private static extern bool CryptSetProvParam(CryptHandle safeProvHandle, CryptProvParam dwParam, [MarshalAs(UnmanagedType.LPArray)] byte[] pbData, SecurityInformationFlags dwFlags);
+
+
+        CryptHandle handle;
+        bool ncrypt = false;
+        public CertAclHelper(X509Certificate2 certificate)
+        {
+            CryptHandle certPkeyHandle;
+            KeySpec keySpec;
+
+            bool ownHandle;
+            if (CryptAcquireCertificatePrivateKey(
+                    certificate.Handle,
+                    (uint)CryptAcquireKeyFlags.CRYPT_ACQUIRE_SILENT_FLAG | (uint)CryptAcquireKeyFlagControl.CRYPT_ACQUIRE_ALLOW_NCRYPT_KEY_FLAG,
+                    IntPtr.Zero,
+                    out certPkeyHandle,
+                    out keySpec,
+                    out ownHandle))
+            {
+                if (!ownHandle)
+                {
+                    throw new NotSupportedException("Could not take ownership of certificate private key handle");
+                }
+
+                if (keySpec == KeySpec.CERT_NCRYPT_KEY_SPEC)
+                    ncrypt = true;
+                handle = certPkeyHandle;
+            }
+            else
+            {
+                throw new Win32Exception(Marshal.GetLastWin32Error());
+
+            }
+        }
+
+
+        public FileSecurity Acl
+        {
+            get
+            {
+
+                SafeSecurityDescriptorPtr securityDescriptorBuffer;
+                uint securityDescriptorSize = 0;
+                if (ncrypt)
+                {
+                    if (NCryptGetProperty(
+                        handle,
+                        "Security Descr",
+                        new SafeSecurityDescriptorPtr(),
+                        0,
+                        ref securityDescriptorSize,
+                        SecurityInformationFlags.DACL_SECURITY_INFORMATION) != 0)
+                    {
+                        throw new Exception("Could not get security descriptor");
+                    }
+                    securityDescriptorBuffer = new SafeSecurityDescriptorPtr(securityDescriptorSize);
+
+                    if (NCryptGetProperty(
+                        handle,
+                        "Security Descr",
+                        securityDescriptorBuffer,
+                        securityDescriptorSize,
+                        ref securityDescriptorSize,
+                        SecurityInformationFlags.DACL_SECURITY_INFORMATION) != 0)
+                    {
+                        throw new Exception("Could not get security descriptor");
```

Comment:
Same here, I believe you can use the return value from `NCryptGetProperty` to derive the Win32 error message for the failure.